include/circt/Dialect/FIRRTL/InstanceGraph.h

```diff
@@ -0,0 +1,148 @@
+//===- InstanceGraph.h - Instance graph -------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the FIRRTL InstanceGraph, which is similar to a CallGraph.
+//
+//===----------------------------------------------------------------------===//
+#ifndef CIRCT_DIALECT_FIRRTL_INSTANCEGRAPH_H
+#define CIRCT_DIALECT_FIRRTL_INSTANCEGRAPH_H
+
+#include "circt/Dialect/FIRRTL/FIRRTLOps.h"
+#include "circt/Support/LLVM.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/OperationSupport.h"
+#include "llvm/ADT/GraphTraits.h"
+#include "llvm/ADT/SmallVector.h"
+#include <memory>
+
+namespace circt {
+namespace firrtl {
+
+class InstanceGraphNode {
+  using EdgeSetT = llvm::SmallVector<InstanceGraphNode *, 1>;
```

Comment:
I'd go with just `EdgeSet` or `EdgeArray` or `EdgeList`, dropping the T.  The capitalization approach in MLIR eliminates the need for _t convention.

Why an inline size of 1?  These nodes are all heap allocated, and I doubt that 1 really captures most cases.

Reply:
Switched it to a std::vector using `EdgeVec`.